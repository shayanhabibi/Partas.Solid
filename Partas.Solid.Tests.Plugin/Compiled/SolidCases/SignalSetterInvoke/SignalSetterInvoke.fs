module Partas.Solid.Tests.SolidCases.SignalSetterInvoke

open Partas.Solid
open Fable.Core
open Fable.Core.JS

[<Erase>]
type WordRotate() =
    inherit div()
    [<Erase>] member val words: string[] = unbox null with get,set
    [<Erase>] member val duration: int = unbox null with get,set
    [<SolidTypeComponentAttribute>]
    member props.constructor =
        let index,setIndex = createSignal(0)
        createEffect(
            fun () ->
                let interval = setInterval (fun () -> setIndex.Invoke(fun prevIndex -> (prevIndex + 1) % (props.words.Length))) props.duration
                onCleanup(fun () -> clearInterval(interval))
        )
        div()