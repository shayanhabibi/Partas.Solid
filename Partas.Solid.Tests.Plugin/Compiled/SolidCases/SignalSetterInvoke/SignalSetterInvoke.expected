import { onCleanup, createEffect, createSignal, splitProps } from "solid-js";

export function WordRotate(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["words", "duration"]);
    const patternInput = createSignal(0);
    createEffect(() => {
        const interval = setInterval(() => {
            patternInput[1]((prevIndex) => (((prevIndex + 1) % PARTAS_LOCAL.words.length) | 0));
        }, PARTAS_LOCAL.duration) | 0;
        onCleanup(() => {
            clearInterval(interval);
        });
    });
    return <div />;
}

